### PR TITLE
CMK support for VMware migration and addition of few properties for VMware DR monitoring

### PR DESCRIPTION
**Pull request by anmolbhatia289**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
There is lots of added properties and they seem not READONLY. Please check for backward-compatibility.

Also I do not see any corresponding change in example.

**Comment by anmolbhatia289** [comment]
Added "readOnly" to newly added properties. Added properties in required examples. 

The following properties: 

- ProcessServer.health
- ProcessServer.psStatsRefreshTime
- ProcessServer.throughputUploadPendingDataInBytes
- ProcessServer.throughputInMBps
- ProcessServer.throughputInBytes
- ProcessServer.throughputStatus
- ProcessServer.marsCommunicationStatus
- ProcessServer.marsRegistrationStatus

are part of VMware specific fabric details. Since, all the fabric examples have HyperV fabric details and not VMware, these properties are not required to be added in examples. 

Additionally, 

InMageAzureV2ReplicationDetails.processServerName is also not required since all the protected item related examples have HyperV specific protected item details and not InMageAzureV2.

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json [hunk 1]
```diff
@@ -16176,6 +16176,10 @@
             "name": "DiskAccountType",
             "modelAsString": true
           }
+        },
+        "diskEncryptionSetId": {
```
Comment (ryansbenson) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json]: This appears to be the only non-readonly field your adding which constitutes a breaking change and requires a new api version.

See 'New property added to response'
https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo

I see you have the response with this field marked as readonly but it still breaks the get-put scenario called out in the doc.

**Comment by anmolbhatia289** [comment]
> ### azure-sdk-for-net
> `❌` failed [[Logs](https://sdkautomation.blob.core.windows.net/sdkautomation-pipeline/Azure/azure-rest-api-specs/8390/azure-sdk-for-net/logs.txt)] [Expand Details]

The failure here is due to a ReplicationEligibilityResults API. The changes made in this PR are not related to this API or any of the related models.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json (hunk 1 above)
Comment (anmolbhatia289) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json]: As this is an optional input property and which cannot be updated once set, won’t this be safe in the PUT -> GET -> PUT scenario? From service side, we retain the existing value of this property if a new PUT comes where this value is not set (the property is tagged as “read-only” in the response).

Comment (weidongxu-microsoft) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json]: There is [x-ms-mutability](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-mutability) which might help in this situation (Though I am not sure how different code generator would handle it. Well at least from reader the intention would be clearer).

I guess we know many service actually use merge in PUT to handle backward compatibility. But that is really a service by service (or even endpoint by endpoint?) solution, so I think ARM had to ask.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side since this is a property which can be set only once. Hopefully it doesn't break the template idempotency. Re-PUtting with the same value should work.

**Pull request merged**